### PR TITLE
🎉 Source Amazon Seller Partner - Implement reportOptions for all missing reports

### PR DESCRIPTION
**Pull request by animer3009**
## What
Implement reportOptions for all missing reports and refactor

## How
With changing Amazon Seller Partner Source connector

Issue: https://github.com/airbytehq/airbyte/issues/23604

[Terminal Saved Build Output.txt](https://github.com/airbytehq/airbyte/files/10860146/Terminal.Saved.Build.Output.txt)
[Terminal Saved Integration Output.txt](https://github.com/airbytehq/airbyte/files/10860147/Terminal.Saved.Integration.Output.txt)
[Terminal Saved unit_test Output.txt](https://github.com/airbytehq/airbyte/files/10860148/Terminal.Saved.unit_test.Output.txt)

### CODE REVIEW

**Comment by animer3009** [comment]
@lazebnyi @bazarnov @davydov-d 
Guys please review my changes to merge.

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py [hunk 1]
```diff
@@ -615,45 +580,42 @@ def _report_data(
     ) -> Mapping[str, Any]:
         data = super()._report_data(sync_mode, cursor_field, stream_slice, stream_state)
         options = self.report_options()
-        if options is not None:
+        if options.get("reportPeriod") is not None:
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py:583]: Despite your changes, `options` can still be `None` and we should handle it
```suggestion
        if options and options.get("reportPeriod") is not None:
```

**Review by davydov-d — COMMENTED** [review]
@animer3009 thanks for your contribution! I have one minor comment, otherwise it looks good. We should be good to proceed when the tests pass

**Comment by animer3009** [comment]
Thank you @davydov-d .
I made commit of the change.

**Review by davydov-d — APPROVED** [review]

**Comment by davydov-d** [comment]
@animer3009 no. We have some issues running Github workflows against contributors PRs. I'll merge this PR once we have this issue resolved and the checks go green

**Pull request merged**